imagetest/test_suites/packagevalidation/package_test.go

```diff
@@ -151,15 +172,45 @@ func TestGuestPackages(t *testing.T) {
 		return
 	}
 
-	missingPkgs := findUniq(requiredPkgs, installedPkgs)
-	for _, pkg := range missingPkgs {
-		t.Errorf("Required package not installed: %s\n", pkg)
+	installedMap := make(map[string]bool)
+	for _, curr := range strings.Split(string(out), "\n") {
+		installedMap[curr] = true
 	}
 
-	shouldnotPkgs := findUniq(excludePkgs, installedPkgs) // find whats installed
-	shouldnotPkgs = findUniq(excludePkgs, shouldnotPkgs)  // it shouldn't be
-	for _, pkg := range shouldnotPkgs {
-		t.Errorf("Package installed but should not be: %s\n", pkg)
-	}
+	for _, curr := range pkgs {
+		skipPackage := false
+		for _, skipExpression := range curr.imagesSkip {
+			if strings.Contains(image, skipExpression) {
+				skipPackage = true
+				break
+			}
+		}
+
+		imageMatched := len(curr.images) == 0
+		for _, matchExpression := range curr.images {
+			if strings.Contains(image, matchExpression) {
+				imageMatched = true
+			}
+		}
+
+		if skipPackage || !imageMatched {
+			break
```

Comment:
`s/continue/break/`, no? This way we continue to the next package rather than skipping the rest of them.

Reply:
exactly, good catch. thank you.

Reply:
done. thank you.